### DISCUSSION TITLE
nwg-shell hard & optional dependencies

### DISCUSSION THREAD

**Discussion by Antiz96** [nwg-piotr/nwg-shell#393] [General]
Hey :wave: 

Someone opened [a ticket on the Arch side](https://gitlab.archlinux.org/archlinux/packaging/packages/nwg-shell/-/issues/1) being confused about nwg-shell hard & optional dependencies (e.g. why is `foot` a hard dependency and not an optional one as "suggested terminal emulator", like `thunar` is for file manager).

From what I understood,  `sway` is actually required (despite `nwg-shell` working with other wayland compositor, such as `hyprland`). But I'd like your confirmation for the rest.

For "third-parties" [dependencies](https://gitlab.archlinux.org/archlinux/packaging/packages/nwg-shell/-/blob/main/PKGBUILD?ref_type=heads#L11-19) (meaning outside of the nwg-* ones), could you confirm me which ones are actually not **hardly required** and could eventually be "demoted" to optional dependencies, e.g. as suggested choices (if there are)?

Thanks in advance :pray:

**Comment by nwg-piotr** [comment]
Hello,

The shell has been developed for sway, and optionally works with Hyprland. Other compositors **are not supported** (except for derivatives like swayfx, and - they say - sway-compatible like miracle). Installing nwg-shell ~~components~~ w/o sway makes no sense, so this must remain a hard dependency.

If it comes to foot, it's used by default by launchers. Use of other terminal emulators is possible with the `-term <string>`, but nwg-shell-config does not support this flag.

[edit]
- it seems ttf-dejavu is unnecessary, as it's a part of ttf-font required by sway
- python-dasbus is required by nwg-panel, so may be removed here
- python-geopy is required by nwg-shell-config
- python-yaml is optional for azote and used nowhere else, if I remember well

Actually, the meta-package purpose is to install a ready-for-use product. If someone wants it to consist of different components, let them feel free to give up on the nwg-shell-package and install stuff one by one.

**Comment by Antiz96** [comment]
Alright, thanks for the confirmation! I'll close the ticket on the Arch side.

**Comment by Antiz96** [comment]
Replying to the above part of the EDIT, which I only saw right now: 

>    it seems ttf-dejavu is unnecessary, as it's a part of ttf-font required by sway
>    python-dasbus is required by nwg-panel, so may be removed here
>    python-geopy is required by nwg-shell-config
>    python-yaml is optional for azote and used nowhere else, if I remember well

Alright, thanks.

> Actually, the meta-package purpose is to install a ready-for-use product. If someone wants it to consist of different components, let them feel free to give up on the nwg-shell-package and install stuff one by one.

Yeah I agree, makes sense. I'll close the ticket with that ;)

**Reply by Antiz96** [reply]
I [closed the Arch ticket](https://gitlab.archlinux.org/archlinux/packaging/packages/nwg-shell/-/issues/1#note_209584) and [removed dependencies already pulled by other nwg packages](https://gitlab.archlinux.org/archlinux/packaging/packages/nwg-shell/-/commit/7807270a16b74f60df9f9f7b4b512c570608b0d6).

Feel free to close this discussion :)
